### PR TITLE
Xsoar saas upload flow test

### PR DESCRIPTION
**Pull request by omerKarkKatz**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-9144)

## Description
A test upload for the xsoar_saas mp to check that a pack that is relevant only to xsoar_saas mp goes only to the correct xsoar_saas mp. 
This is to prevent potential future changes in artifact files that will lead mixing around between marketplaces.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by yaakovpraisler — COMMENTED** [review]
Looks good, see my notes -

File: Utils/test_upload_flow/verify_bucket.py [hunk 1]
```diff
@@ -46,13 +54,13 @@ def wrapper(self, *args, **kwargs):
...
         try:
-            result, pack_id = func(self, *args, **kwargs)
+            result, pack_id, message = func(self, *args, **kwargs)
             self.is_valid = self.is_valid and result
 
             if not result:
                 raise Exception(f"Failed when running validation - {func.__name__}")
 
-            logging.info(f"[{pack_id}] Successfully {MSG_DICT[func.__name__]}")
+            logging.info(f"[{pack_id}] Successfully {message}")
```
[the remark this answers is not in the document]
Comment (omerKarkKatz) [reply Utils/test_upload_flow/verify_bucket.py:63]: I think it was not designed well for reusing the messages or using the messages with different arguments (placeholders) as used now in verify_pack_not_in_marketplace and verify_pack_in_marketplace

File: Utils/test_upload_flow/verify_bucket.py [hunk 2]
```diff
@@ -197,15 +205,28 @@ def verify_new_pack(self, pack_id, pack_items):
...
         """
         version_exists = [self.gcp.is_in_index(pack_id), self.gcp.download_and_extract_pack(pack_id, '1.0.0')]
-        return not any(version_exists), pack_id
+        return (not any(version_exists),
+                pack_id,
+                MSG_DICT['verify_pack_not_in_marketplace'].format(pack_id, marketplace)
+                )
+
+    @logger
+    def verify_pack_in_marketplace(self, pack_id, marketplace, version, pack_items):
```
[the remark this answers is not in the document]
Comment (omerKarkKatz) [reply Utils/test_upload_flow/verify_bucket.py:222]: ok, i can do that no problem. I

[the remark this answers is not in the document]
Comment (omerKarkKatz) [reply Utils/test_upload_flow/verify_bucket.py:222]: I decided to extract this functionality to a different function I don't think that the RN logic is related to checking if a content pack exists in the marketplace and future tests don't need to start thinking about handling the RN if it's not the purpose of the test.

[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Utils/test_upload_flow/verify_bucket.py:226]: I do think that a part of checking if a new pack was added to a bucket, is to check if its changelog is generated correctly.

File: Utils/test_upload_flow/verify_bucket.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Utils/test_upload_flow/verify_bucket.py:63]: You can get the `pack_id` and `marketplace` params of the func from the wrapper `*args`

**Pull request closed without merging**